src/containers.js

```diff
@@ -38,19 +42,72 @@ const update = async (fileName, fileContents, install = false) => {
     return false;
   }
 
-  // write temp file, validate and then overwrite
   if (!await validComposeFile(fileContents)) {
     throw new Error(`Invalid docker-compose yml for file ${fileName}`);
   }
 
+  if (backupIsEnabled()) {
+    await backupYmlFiles();
+  }
+
   await overwriteComposeFile(filePath, fileContents);
   await dockerComposeCli.pull(filePath);
 
   return true;
 };
 
+
+/**
+ * Generates a new name containing date of creation and random characters to avoid most collisions
+ * @returns {string} Name for folder e.g. 2023-07-17_131438_817_2Tjx
+ */
+const genBackupFolderName = () => {
+  const filefriendlyDatetime = new Date()
+    .toISOString()
+    .replace('T', '_')
+    .replaceAll(':', '')
+    .replace('.', '_')
+    .replace('Z', ''); // e.g. 2023-07-17_131438_817
+
```

Comment:
> Can this big chain be replaced with a single one with a regex?

It would have to be two lines unless we decide to always replace with the same character
```suggestion
    .replaceAll(/[T.]/g, '_')
    .replaceAll(/[Z:]/g, ''); // e.g. 2023-07-17_131438_817
```
```suggestion
    .replaceAll(/[T:.Z]/g, '_'); // e.g. 2023-07-17_13_14_38_817_
```

I find the first regexp here a bit less clear and prefer the original format, but am open for either

Reply:
PR updated using dashes, new format `2023-07-17-13-14-38-817-2Tjx`